pdl_syntax.md

```diff
@@ -0,0 +1,819 @@
+
+#### Keyword Escaping
+
+```
+namespace com.linkedin.pegasus.generator.test.idl.escaping
+
+record PdlKeywordEscaping {
+  `namespace`: string
+  `record`: string
+  `null`: string
+  `enum`: string
+  recordName: record `record` { }
+}
+```
+
+#### Namespace/Package escaping
+
+```
+namespace com.linkedin.pegasus.generator.test.idl.escaping.`record`
+package com.linkedin.pegasus.generator.test.idl.escaping.override.`typeref`
+
+/**
+ * Ensures that the namespace and package are properly escaped at the root as well as in scoped named-type declarations.
+ */
+record NamespacePackageEscaping {
+  x: {
+    namespace com.x.y.z.`enum`
+    package com.a.b.c.`fixed`
+
+    record Foo {}
+  }
+}
+```
+
+#### Property key escaping 
+
+If you want Pegasus to treat property key name with dots as one string key, please use backticks to escape such string.
+For example: 
+
+```
+namespace com.linkedin.pegasus.generator.test.idl.escaping
+
+record PropertyKeyEscaping {
+  @`namespace` = "foo.bar"
+  @`test.path` = 1
+  @validate.`com.linkedin.CustomValidator` = "foo"
+  aField: string
+}
+```
+
+
+## Primitive Types
+
+The Pegasus primitive types are : int, long, float, double, boolean, string and bytes.
+
+For Example:
+```
+namespace com.linkedin.pegasus.generator.test.idl.records
+
+record WithPrimitives {
+  intField: int
+  longField: long
+  floatField: float
+  doubleField: double
+  booleanField: boolean
+  stringField: string
+  bytesField: bytes
+}
+```
+
+Primitive types with default values:
+```
+namespace com.linkedin.pegasus.generator.test.idl.records
+
+record WithPrimitiveDefaults {
+  intWithDefault: int = 1
+  longWithDefault: long = 3000000000
+  floatWithDefault: float = 3.3
+  doubleWithDefault: double = 4.4E38
+  booleanWithDefault: boolean = true
+  stringWithDefault: string = "DEFAULT"
+  bytesWithDefault: bytes = "\u0007"
+}
+```
+
+
+## Array Type
+
+Pegasus Arrays are defined as a collection of a particular "items" type.
+
+For Example: 
+```
+namespace com.linkedin.pegasus.generator.test.idl.arrays
+
+record WithPrimitivesArray {
+  ints: array[int]
+  longs: array[long]
+  floats: array[float]
+  doubles: array[double]
+  booleans: array[boolean]
+  strings: array[string]
+  bytes: array[bytes]
+}
+```
+
+Primitive arrays with default values: 
+```
+namespace com.linkedin.pegasus.generator.test.idl.arrays
+
+record WithPrimitivesArrayDefaults {
+  ints: array[int] = [1]
+  longs: array[long] = [3000000000]
+  floats: array[float] = [3.3]
+  doubles: array[double] = [4.4E38]
+  booleans: array[boolean] = [true]
+  strings: array[string] = ["hello"]
+  bytes: array[bytes] = ["\u0007"]
+}
+```
+
+Record arrays:
+```
+namespace com.linkedin.pegasus.generator.test.idl.arrays
+
+import com.linkedin.pegasus.generator.test.idl.enums.Fruits
+import com.linkedin.pegasus.generator.test.idl.records.Empty
+
+record WithRecordArray {
+  empties: array[Empty]
+  fruits: array[Fruits]
```

Comment:
Fruits here is enum, not a record, I think you can change the text to say "Record or Enum arrays:"

Reply:
Fixed